Python/pystate.c

```diff
@@ -1388,6 +1389,10 @@ PyThreadState_Next(PyThreadState *tstate) {
 PyObject *
 _PyThread_CurrentFrames(void)
 {
+    // Disable the GC as this can cause a deadlock the interpreter.
+    // See issues 116969 and 106883.
+    PyGC_Disable();
```

Comment:
Save the old state with something like:

```
int gc_was_enabled = PyGC_Disable();
```

And move this down after the things that may `return NULL`.

Reply:
Done, thanks!